eth/filters/trace_api.go

```diff
@@ -58,6 +59,17 @@ func (api *FilterAPI) NewPendingTransactionsWithTrace(ctx context.Context, trace
 			msg     *core.Message
 		)
 
+		metricsPendingTxsNew.Inc()
+		defer metricsPendingTxsEnd.Inc()
+
+		// Recover from any panics. Should be the last deferred call so it runs
+		// first.
+		defer func() {
```

Comment:
Where do we expect any panics to happen?

Reply:
We don't expect a panic any more than any other bugs, and all currently known vectors for a panic have been fixed, but if we ever do introduce or find a panic it's preferable to recover instead of allowing the process to crash. Crashing is expensive in terms of time to start up and re-peer so clients can't immediately reconnect like they can with other websocket failures.